### DISCUSSION TITLE
Python und Boolean

### DISCUSSION THREAD

**Discussion by hoetzgit** [roe-dl/weewx-DWD#18] [Q&A] [answered]
Hallo, ich probiere etwas mit dem "uba Mod", aber meine Python Kenntnisse reichen nicht.

Bei Aufruf:
`/usr/local/bin/weewx-DWD/dwd-mosmix --config=/home/weewx/weewx.conf --print-uba=aqi --dry-run 509`

kommt der JSON String immer **zusätzlich** mit der Ausgabe:
```
successfully downloaded https://www.umweltbundesamt.de/api/air_data/v2/airquality/json?date_from=2022-03-10&date_to=2022-03-11&lang=de&station=509
[
    {
...
```

Da ich diese Ausgabe für meinen Zweck in diesem Moment nicht möchte, habe ich kein `--verbose` angegeben und `debug`  mit `0` sowie `log_success` überall mit `False` konfiguriert. Zum Eingrenzen des "Problems" habe ich den Code in `download_uba` testweise mit zwei prints so verändert:

```
...
        if reply.status_code==200:
            print('self.log_success is ', self.log_success)
            print('self.verbose is ', self.verbose)
            if self.log_success or self.verbose:
                loginf('successfully downloaded %s' % reply.url)
...
```

Ausgabe mit beiden Werten False aber trotzdem `successfully ...`:
```
self.log_success is  False
self.verbose is  False
successfully downloaded https://www.umweltbundesamt.de/api/air_data/v2/airquality/json?date_from=2022-03-10&date_to=2022-03-11&lang=de&station=509
[
    {
...
```

Ändere ich aber zum Test mal so und setze die Variablen:
```
...
        if reply.status_code==200:
            self.log_success = False
            self.verbose = False
            print('self.log_success is ', self.log_success)
            print('self.verbose is ', self.verbose)
            if self.log_success or self.verbose:
                loginf('successfully downloaded %s' % reply.url)
...
```

dann ist die Ausgabe wie erwartet **ohne** `successfully ...`:
```
self.log_success is  False
self.verbose is  False
[
    {
...

```

Kann es sein, dass z.B. diese Zuweisung:
`self.log_success = forecast_dict.get('log_success',config_dict['DeutscherWetterdienst'].get('log_success',config_dict.get('log_success',False)))
`
keinen "echten" Boolean erzeugt?

Viele Grüße und ein schönes Wochenende
Henry

**Ergänzung**:
ist in der `weewx.conf` `log_success` nirgends definiert, ist die Ausgabe auch korrekt, da dann der default `False` in der Zuweisung den Boolean offensichtlich korrekt setzt.

**Comment by roe-dl** [comment, marked as answer]
So, wie sich das anhört, wäre dieses Thema wohl unter den Issues gut untergebracht. Ich muß es mir ansehen. Ausgeschlossen ist das nicht.

**Comment by roe-dl** [comment]
Es handelte sich in der Tat um einen Bug, siehe #19

commit 6c18d1cc5a7a40fe64c6511701115b8f59f7d9e1
commit c34bc86f0ef601b81dad747eadc91d8fa0f48f2e
